nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/main/java/org/apache/nifi/util/db/JdbcCommon.java

```diff
@@ -711,18 +711,59 @@ public static void setParameters(final PreparedStatement stmt, final Map<String,
         }
     }
 
+    /**
+     * Sets all of the appropriate parameters on the given PreparedStatement, based on the given FlowFile attributes.
+     *
+     * @param stmt the statement to set the parameters on
+     * @param attributes the attributes from which to derive parameter indices, values, and types
+     * @throws SQLException if the PreparedStatement throws a SQLException when the appropriate setter is called
+     */
+    public static void setSensitiveParameters(final PreparedStatement stmt, final Map<String, SensitiveValueWrapper> attributes) throws SQLException {
+        for (final Map.Entry<String, SensitiveValueWrapper> entry : attributes.entrySet()) {
+            final String key = entry.getKey();
+            final boolean isSensitive = entry.getValue().isSensitive();
+            final String value = entry.getValue().getValue();
+            final String logValue = isSensitive ? "???" : value;
```

Comment:
Recommend creating a static value named `MASKED` and reusing it.
```suggestion
            final String logValue = isSensitive ? "MASKED" : value;
```